server/src/test/java/org/elasticsearch/index/mapper/NativeArrayIntegrationTestCase.java

```diff
@@ -82,6 +81,21 @@ protected void verifySyntheticArray(Object[][] arrays) throws IOException {
         verifySyntheticArray(arrays, mapping, "_id");
     }
 
+    private void verifyResponseArray(Object[] expected, Object actual) {
+        if (expected == null) {
+            assertThat(actual, nullValue());
+        } else if (actual instanceof List<?> actualArray) {
+            if (expected.length == 0) {
+                assertThat(actualArray, empty());
+            } else {
+                assertThat(actualArray, Matchers.contains(expected));
+            }
+        } else {
```

Comment:
Had to update this check since single-value arrays are unwrapped to single-values

Reply:
This was also part of the change removing the check for context.isImmediateParentAnArray(), since the change caused single-slot arrays to be returned as single values.

Since we're rethinking that solution, I'll probably end up reverting this too.